src/api/bookmark-service/bookmark.ts

```diff
@@ -32,28 +10,79 @@ const instance = createHttpClient()
   .setBaseUrl(import.meta.env.VITE_API_BASE_URL)
   .build();
 
-const mapBookmark = (bookmarkResponse: any) => {
-  return bookmarkResponse;
+const mapTag = (tagList: Tag[]) => {
+  return tagList;
+};
+
+const mapBookmark = (bookmarkResponse: Bookmark[]) => {
+  const defaultTagName = '전체';
+  const result: BookmarkByTagName = {
+    [defaultTagName]: {
+      tagId: -1,
+      bookmarkList: [],
+    },
+  };
+
+  bookmarkResponse.forEach((bookmark) => {
+    result[defaultTagName].bookmarkList.push(bookmark);
+    bookmark.tags.forEach((tag) => {
+      if (tag in result) {
+        result[tag].bookmarkList.push(bookmark);
+      } else {
+        result[tag] = {
+          tagId: -1,
+          bookmarkList: [bookmark],
+        };
+      }
+    });
+  });
+
+  return result;
 };
 
 const dispatchError = (err: ErrorResponse) => {
-  // Error를 한 곳에 모아놓고 커스텀으로 정의하여 사용하면 좋을 것 같습니다.
   if (err.errorCode === 'AUTH_INVALID_TOKEN') {
     throw new Error('TOKEN_EXPIRED_ERROR');
   } else {
     throw new Error('UNKNOWN_ERROR');
   }
 };
 
-const getBookmarkList = (): Promise<Bookmark[]> => {
-  const params = {
-    page: 0,
-    size: 100,
-  };
+const getTagList = ({ page = 0, size = 100 } = {}): Promise<Tag[]> => {
+  const params = { page, size };
+  return instance
+    .get('/bookmark/tags', { params })
+    .then((res) => mapTag(res as Tag[]))
```

Comment:
```suggestion
  return instance
    .get<Promise<Tag[]>>('/bookmark/tags', { params })
    .then((res) => mapTag(res))
```

응답값 Map 함수에 사용되는 type assertion 부분을 없애보려고 했는데,
Infra쪽 코드를 훑어봐야될 것 같고 의견도 여쭤봐야해서 일단 코멘트만 남겨둡니다..! 😥

Reply:
infra 쪽에서 없애면 될겁니다! 응답값이 추론이 안될거라고 생각해서 넣어놓은 걸로 기억하는데 map 함수에서 처리하면 되겠네요 ㅎㅎ